ios/RNPurchases.m

```diff
@@ -592,3 +599,43 @@ - (NSString *)platformFlavorVersion {
 }
 
 @end
+
+@implementation NSObject (NSNullMapping)
+
+- (id)mappingNSNullToNil {
+    if ([self isKindOfClass:[NSNull class]]) {
+        return nil;
+    } else if ([self isKindOfClass:NSDictionary.class]) {
+        NSMutableDictionary *filteredDict = [NSMutableDictionary dictionary];
+        NSDictionary *originalDict = (NSDictionary *)self;
+
+        for (id key in originalDict) {
+            id value = [originalDict[key] mappingNSNullToNil];
+            if (value) {
+                // Only add non-nil values to the dictionary
```

Comment:
Hmm I wonder if we actually want to keep the values in the dict, but just as `nil`... what do you think? (same for arrays)

Reply:
I thought about that, but then I remembered that we're in Objective-C world. And there I believe you cannot insert `nil` into collections (I believe that might be a reason why `NSNull` exists?). The thing is:
* Setting `nil` for a key in a `NSDictionary` is used to remove the value for that key.
* Setting `nil` in an `NSArray` is not allowed.